shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/GpuHashJoin.scala

```diff
@@ -197,19 +198,21 @@ trait GpuHashJoin extends GpuExec with HashJoin {
       override def hasNext: Boolean = {
         while (nextCb.isEmpty && (first || stream.hasNext)) {
           if (stream.hasNext) {
+            val startTime = System.nanoTime()
             val cb = if (shouldFilterStreamTableForNulls) {
               filterStreamedTable(stream.next())
             } else {
               stream.next()
             }
-            val startTime = System.nanoTime()
```

Comment:
Note that moving `startTime` here is changing the meaning of what `totalTime` was doing before.  Now instead of `totalTime` measuring the time spent doing the join it's also measuring time spent waiting.

Reply:
Yes，I think the total time should include every phase belongs to this node；

And，i see in the `else` branch，the `totalTime` just include the `cb = xxx`

Reply:
Agreed, we should have `totalTime` be the total.  There's a separate `joinTime` that can be used to measure how much time was spend performing the join on the GPU once the necessary batches have arrived.